### DISCUSSION TITLE
pagefind doesn't see my HTML

### DISCUSSION THREAD

**Discussion by NickHodges** [Pagefind/pagefind#608] [General]
So, when I run pagefind on my site, I get  what is below.

However, I assure you that there are *.HTML files in my /dist directory, and that they do indeed have <body> tags with content in them. 

I'm lost.  Can someone help me find what I'm doing wrong?

```
nickhodges@penguin:~/cactus/dist$ npx -y pagefind --site dist --serve

Running Pagefind v1.1.0 (Extended)
Running from: "/home/nickhodges/cactus/dist"
Source:       "dist"
Output:       "dist/pagefind"

[Walking source directory]
Found 0 files matching **/*.{html}

[Parsing files]
Did not find a data-pagefind-body element on the site.
↳ Indexing all <body> elements on the site.

[Reading languages]
Discovered 0 languages: 

[Building search indexes]
Total: 
  Indexed 0 languages
  Indexed 0 pages
  Indexed 0 words
  Indexed 0 filters
  Indexed 0 sorts
Error: Pagefind wasn't able to build an index. 
Most likely, the directory passed to Pagefind was empty or did not contain any html files.
nickhodges@penguin:~/cactus/dist$ 
```

**Comment by bglw** [comment]
👋 @NickHodges 

It looks like you're running from the `dist` folder as a working directory, with a source/output configured as `dist` — so Pagefind will be looking for `/home/nickhodges/cactus/dist/dist/**/*.html`

Try running from one directory up (`/home/nickhodges/cactus`) 🙂

**Reply by NickHodges** [reply]
Well, yep, that work.  Thank you.

**Comment by NickHodges** [comment]
@bglw 

Okay, but then that puts all the output files in my project.  

What I'd like is for pagefind to run on the production site that is indeed put into `/dist` when I build.

**Reply by bglw** [reply]
Hmm I'm not sure I can picture what is running — I would expect

- You're in the `/home/nickhodges/cactus` directory
- You run `npx -y pagefind --site dist`
- It reads the files at, e.g., `/home/nickhodges/cactus/dist/index.html`
- It outputs the index into a folder `/home/nickhodges/cactus/dist/pagefind/`

**Reply by NickHodges** [reply]
All of that is correct. 

It reports this:

```
Running Pagefind v1.1.0 (Extended)
Running from: "/home/nickhodges/cactus/dist"
Source:       "dist"
Output:       "dist/pagefind"
```

Which all looks right to me.  The files are there, including /dist/index.html

Yet it reports it sees nothing.

**Reply by NickHodges** [reply]
Thank you for helping me.

**Reply by bglw** [reply]
Ah I still see `Running from: "/home/nickhodges/cactus/dist"` — so the first bullet isn't `You're in the /home/nickhodges/cactus directory` no? You're in the `/home/nickhodges/cactus/dist` directory

**Reply by NickHodges** [reply]
Okay, this got it running.  And deployed.  But it doesn't want to work from my deployed site.  I'll post another query.
